backends/p4tools/modules/testgen/targets/bmv2/test_backend/protobuf_ir.cpp

```diff
@@ -140,6 +164,105 @@ entities {
     return TEST_CASE;
 }
 
+std::string ProtobufIr::formatNetworkValue(const std::string &type, const IR::Expression *value) {
+    if (type == "hex_str") {
+        return formatHexExpr(value);
+    }
+    // At this point, any value must be a constant.
+    const auto *constant = value->checkedTo<IR::Constant>();
+    if (type == "ipv4") {
+        auto convertedString = convertToIpv4String(convertBigIntToBytes(constant->value, 32));
+        if (convertedString.has_value()) {
+            return convertedString.value();
+        }
+        BUG("Failed to convert IPv4 string.");
```

Comment:
can we include the value that failed to be converted here? Similarly a few more times below.